### DISCUSSION TITLE
Field for noting the intended field of use of a software?

### DISCUSSION THREAD

**Discussion by Ainali** [publiccodeyml/publiccode.yml#111] [Ideas]
I spotted the [Classification of the Functions of Government](https://unstats.un.org/unsd/classifications/Econ/Structure) (COFOG) at the United Nations Statistics Division and got the idea that this could be a field in the publiccode.yml standard.

It would allow for filtering in a catalog on another dimension, one that perhaps is quite relevant to a civil servant, since they probably know which field they are in and looking for software in. It would also allow for another type of overview/statistics and answer questions like “Are there any parts of the government that still doesn’t have any open source software?”

Using the COFOG for this makes sense since this is used for reporting expenditure, so each agency should already be classified using this framework. Hence, the added work to determine this should be low.

The system has 188 classes in three levels, and I suggest that any of these should be a valid value for the field and that the field should allow for multiple values (since a codebase may do more than one thing).

**Comment by bzg** [comment]
Hi Jan,

> I spotted the Classification of the Functions of Government (COFOG) at the United Nations Statistics Division and got the idea that this could be a field in the publiccode.yml standard.

Thanks for sharing this!

> Using the COFOG for this makes sense since this is used for reporting expenditure, so each agency should already be classified using this framework. Hence, the added work to determine this should be low.

I'm curious if it actually works.  I'm not sure there is a direct mapping between "functions" and "structures" - I'd be happy to be wrong and to test this, though.

**Reply by Ainali** [reply]
> I'm curious if it actually works. I'm not sure there is a direct mapping between "functions" and "structures" - I'd be happy to be wrong and to test this, though.

I'm not sure what you mean by "works". If you are asking if the countries are able to classify agencies on what function they have? Yes, they have been doing so for years. Or do you have doubts it is possible to map to software? Well, there I am not 100% sure either, but I believe that the categories are defined well enough so that one can understand the purpose of them and see if they are suitable to map onto a codebase. Sure, there might be corner cases, but I imagine that by adding one extra classification or moving up one step in the hierarchy, that will be solved.

**Comment by bzg** [comment]
> I'm not sure what you mean by "works". If you are asking if the countries are able to classify agencies on what function they have?

My understanding is that COFOG is used to classify expenditures of the public agencies, not the agencies themselves.

> Yes, they have been doing so for years. 

Do you have a URL where I can see how a country classifies its public agencies (not just public expenditures) using COFOG classification?

I've been quickly looking for French examples but could not find any.

My guess is that one agency can spend on multiple COFOG catégories, so the mapping is probably not straightforward.

**Comment by Ainali** [comment]
Yes, it is meant for expenditures, but I understand it as countries lump everything from an agency into one class (but I might be wrong here). 

As the reporting itself is expenditure per class, it has been hard to find the raw data in public or open datasets. So far I have found easy data for [Sweden](https://www.statskontoret.se/contentassets/73dfde44d5bc4cffaace77dd47226c69/arsarbetskrafter-i-staten-oppna-data_webben.xlsx) and some for [Norway](https://www.nsd.no/polsys/data/forvaltning/cofog) (only top level), and possibly UK and Ghana (but I don't have the links at hand).

**Comment by ruphy** [comment]
Hi Jan,

thanks for your proposal!!

As far as Italy is concerned I also can't find any similar grouping of agencies and I share @bzg concerns. I'd like to learn more but the link you posted on Sweden returns a 404 and Norway somehow seems to appear to use a different definition of COFOG for agencies, if my translator is not misleading me (01 is General public services, not 01 Agriculture).

On the merit of the proposal, I am not sure how this classification would help. The majority of software we currently have on a catalogue and the one I expect to grow the most is very generic and not part of a niche, or it's extendable and adaptable to specific needs through business rules (think of a general purposes CMS with special templates, a cloud storage, a city management application or an administrative software). It's also the one with the highest potential for an ecosystem to grow and would easily span all categories.

**Reply by Ainali** [reply]
> Sweden returns a 404

Here is an [archived link](https://web.archive.org/web/20210810191757/https://www.statskontoret.se/contentassets/73dfde44d5bc4cffaace77dd47226c69/arsarbetskrafter-i-staten-oppna-data_webben.xlsx).

> Norway somehow seems to appear to use a different definition of COFOG for agencies, if my translator is not misleading me (01 is General public services, not 01 Agriculture)

I think your translator is wrong, they do use General public services (Alminnelig offentlig tjenesteyting) as 01.

> On the merit of the proposal, I am not sure how this classification would help

That might be true of what is in there today. But there are a lot of potential for software produced by public organizations that are more niche, and that would be easier to discover if there are ways to find them that makes sense. That said, I am not an end user, and it might make sense to close this issue until there is an actual demand unless you agree it would help you now.

**Comment by Clarisse-M-G** [comment]
Hello Jan, Bastien and Riccardo

Actually, I find this discussion highly interesting, not just for visibility of code usage in different fields but, more so, to foster the re-use of existing (often publicly funded) solutions across local agencies and help building communities around these solutions. 

So I would like to ask: would you also see the advantage of including linked standard code lists ( be it the UN [Classification of the Functions of Government COFOG](https://unstats.un.org/unsd/classifications/Econ/Structure) , the European [Core Public Service Vocabulary Application Profile (CPSV-AP) | ISA²](https://ec.europa.eu/isa2/solutions/core-public-service-vocabulary-application-profile-cpsv-ap_en/) or the national (German) [services catalogue](https://fimportal.de/kataloge) )?

Similar to the [European Digital Gateway](https://single-market-economy.ec.europa.eu/document/download/[02134300-8b4e-4d2e-96d0-680cc9208f91_en](https://single-market-economy.ec.europa.eu/document/download/02134300-8b4e-4d2e-96d0-680cc9208f91_en?filename=EC%20Crawler%20documentation_13_08_2025.pdf)?filename=EC%20Crawler%20documentation_13_08_2025.pdf) ?
